web/grab_to_pan.js

```diff
@@ -14,97 +14,91 @@
  * limitations under the License.
  */
 
-/**
- * Construct a GrabToPan instance for a given HTML element.
- * @param options.element {Element}
- * @param options.ignoreTarget {function} optional. See `ignoreTarget(node)`
- * @param options.onActiveChanged {function(boolean)} optional. Called
- *  when grab-to-pan is (de)activated. The first argument is a boolean that
- *  shows whether grab-to-pan is activated.
- */
-function GrabToPan(options) {
-  this.element = options.element;
-  this.document = options.element.ownerDocument;
-  if (typeof options.ignoreTarget === "function") {
-    this.ignoreTarget = options.ignoreTarget;
-  }
-  this.onActiveChanged = options.onActiveChanged;
-
-  // Bind the contexts to ensure that `this` always points to
-  // the GrabToPan instance.
-  this.activate = this.activate.bind(this);
-  this.deactivate = this.deactivate.bind(this);
-  this.toggle = this.toggle.bind(this);
-  this._onmousedown = this._onmousedown.bind(this);
-  this._onmousemove = this._onmousemove.bind(this);
-  this._endPan = this._endPan.bind(this);
-
-  // This overlay will be inserted in the document when the mouse moves during
-  // a grab operation, to ensure that the cursor has the desired appearance.
-  const overlay = (this.overlay = document.createElement("div"));
-  overlay.className = "grab-to-pan-grabbing";
-}
-GrabToPan.prototype = {
+// Class name of element which can be grabbed.
+const CSS_CLASS_GRAB = "grab-to-pan-grab";
+
+class GrabToPan {
   /**
-   * Class name of element which can be grabbed
+   * Construct a GrabToPan instance for a given HTML element.
+   * @param {Element} options.element
+   * @param {function} [options.ignoreTarget] - See `ignoreTarget(node)`.
+   * @param {function(boolean)} [options.onActiveChanged} - Called when
+   *   grab-to-pan is (de)activated. The first argument is a boolean that
+   *   shows whether grab-to-pan is activated.
    */
-  CSS_CLASS_GRAB: "grab-to-pan-grab",
+  constructor(options) {
+    this.element = options.element;
+    this.document = options.element.ownerDocument;
+    if (typeof options.ignoreTarget === "function") {
+      this.ignoreTarget = options.ignoreTarget;
+    }
+    this.onActiveChanged = options.onActiveChanged;
+
+    // Bind the contexts to ensure that `this` always points to
+    // the GrabToPan instance.
+    this.activate = this.activate.bind(this);
+    this.deactivate = this.deactivate.bind(this);
+    this.toggle = this.toggle.bind(this);
+    this._onmousedown = this.#onmousedown.bind(this);
```

Comment:
Wouldn't it be better to call this member variable and the associated method to `onMouseDown` instead (same for `onMouseMove` below) to match the casing in the rest of the code?

Reply:
Yes, that obviously makes more sense. (To be honest, I forgot to look closely at the names and mostly focused on the class-conversion itself.)